adam/curriculum_to_html.py

```diff
@@ -0,0 +1,184 @@
+            # outputtext.append(f"<li>Properties:\n<ul>")
+            # for prop in obj.properties:
+            #    outputtext.append(f"<li>{prop.handle}</li>")
+            # outputtext.append("</ul></li>")
+        outputtext.append("</ul>")
+        if not situation.actions:
+            outputtext.append("<h4>Actions</h4>\n<ul>")
+            for acts in situation.actions:
+                outputtext.append(f"<li>{acts.action_type.handle}</li>")
+            outputtext.append("</ul>")
+        if not situation.relations:
+            outputtext.append("<h4>Relations</h4>\n<ul>")
+            for rel in situation.relations:
+                outputtext.append(
+                    f"<li>{rel.relation_type.handle}({rel.first_slot.ontology_node.handle},{rel.second_slot.ontology_node.handle})</li>"
+                )
+            outputtext.append("</ul>")
+        outputtext.append("</td>")
+        return outputtext
+
+    @staticmethod
+    def _perceptiontext(
+        perception: PerceptualRepresentation[DevelopmentalPrimitivePerceptionFrame]
+    ):
+        """
+        Turns a perception into a list of items in the perceptions frames.
+
+        Given a `PerceptualRepresentation[DevelopmentalPrimitivePerceptionFrame]` the information
+        within is converted into a table entry with headings for the informaiton contained within
+        the example. Returns a List[str]
+
+        """
+        outputtext = [f"<td>"]
+        frame_number = 0
+        for frame in perception.frames:
+            outputtext.append(f"<h4>Frame {frame_number}</h4>")
+            outputtext.append("<h5>Perceived Objects</h5>\n<ul>")
+            for obj in frame.perceived_objects:
+                outputtext.append(f"<li>{obj.debug_handle}</li>")
+            outputtext.append("</ul>")
+            if not frame.property_assertions:
+                outputtext.append("<h5>Property Assertions</h5>\n<ul>")
+                for prop in frame.property_assertions:
+                    outputtext.append(f"<li>{prop}")
+                outputtext.append("</ul>")
+            if not frame.relations:
+                outputtext.append("<h5>Relations</h5>\n<ul>")
+                for rel in frame.relations:
+                    outputtext.append(
+                        f"<li>{rel.relation_type.handle}({rel.arg1},{rel.arg2})"
+                    )
+                outputtext.append("</ul>")
+            frame_number = frame_number + 1
+        outputtext.append("</td>")
+        return outputtext
+
+    @staticmethod
+    def _linguistictext(linguistic: LinearizedDependencyTree):
+        """
+        Parses the Linguistic Description of a Linearized Dependency Tree into a table entry
+
+        Takes a `LinearizedDependencyTree` which is turned into a token sequence and
+        phrased as a sentence for display. Returns a List[str]
+        """
+        outputtext = [f"<td>", " ".join(linguistic.as_token_sequence()), "</td>"]
+        return outputtext
```

Comment:
In this case giving the variable name `outputtext` doesn't help the reader any, so you can just

```python
return [f"<td>", " ".join(linguistic.as_token_sequence()), "</td>"]
```